api4/insights_test.go

```diff
@@ -816,3 +816,87 @@ func TestGetTopThreadsForUserSince(t *testing.T) {
 	require.Nil(t, appErr)
 	require.Len(t, topUser2ThreadsAfterPrivateReplyDelete.Items, 0)
 }
+
+func TestGetTopDMsForUserSince(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	th.ConfigStore.SetReadOnlyFF(false)
+	defer th.ConfigStore.SetReadOnlyFF(true)
+	th.App.UpdateConfig(func(cfg *model.Config) { cfg.FeatureFlags.InsightsEnabled = true })
+	th.App.Srv().SetLicense(model.NewTestLicenseSKU(model.LicenseShortSkuProfessional))
+	// basicuser1 - bu1, basicuser - bu
+	// create dm channels for  bu-bu, bu1-bu1, bu-bu1
+	basicUser := th.BasicUser
+	basicUser1 := th.BasicUser2
+
+	th.LoginBasic2()
+	client := th.Client
+	channelBu1, _, err := client.CreateDirectChannel(basicUser1.Id, basicUser1.Id)
+	require.NoError(t, err)
+
+	th.LoginBasic()
+	client = th.Client
+	channelBu, _, err := client.CreateDirectChannel(basicUser.Id, basicUser.Id)
```

Comment:
Is `channelBu` a direct channel with oneself? If so, should it be included in the results?

Reply:
Yes it is, I'll clean the variable names, I know it's confusing at the moment.

Reply:
Oh, I think it should. But 1/5 and I'm open to correct it, because you, @laneycs , or @matthewbirtch  can help decide  on this better.

1. People talk less with theirselves, except for bookmarks, drafts and such.
2. Couldn't find a reason to exclude.

Reply:
Awesome thank you both :) I'll exclude them and update the tests.

Reply:
This is done.